sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -401,6 +401,25 @@ case class CatalogTable(
     )
   }
 
+  /**
+   * Return the schema binding mode. Defaults to SchemaCompensation if not a view or an older
+   * version, unless the viewSchemaBinding config is set to UNSUPPORTED
+   */
+  def viewSchemaMode: ViewSchemaMode = {
+    if (SQLConf.get.viewSchemaBinding == "UNSUPPORTED") {
+      SchemaUnsupported
+    } else {
+      val schemaMode = properties.getOrElse(VIEW_SCHEMA_MODE, SchemaCompensation.toString)
```

Comment:
Shall we use `SQLConf.get.viewSchemaBinding` as the default value instead of `SchemaCompensation.toString`? If we change the default value of the SQL conf in the future, we may miss updating this one.

Reply:
I was considering that, but the case here is a view which predates the feature.
So if we pick up the config we would have views which behavior changes based on a session config.
That seems weird, even more so if they can refresh themselves.
I'd rather keep the config to only affect CREATE  VIEW.